src/actions/sagemaker/sagemaker_infer.ts

```diff
@@ -0,0 +1,356 @@
+      const s3OutputPath = `s3://${bucket}`
+      winston.debug("s3InputPath", s3InputPath)
+      winston.debug("s3OutputPath", s3OutputPath)
+
+      const transformParams: SageMaker.CreateTransformJobRequest = {
+        ModelName: modelName,
+        TransformJobName: jobName,
+        TransformInput: {
+          DataSource: {
+            S3DataSource: {
+              S3DataType: "S3Prefix", // required
+              S3Uri: s3InputPath, // required
+            },
+          },
+          ContentType: "text/csv",
+        },
+        TransformOutput: {
+          S3OutputPath: s3OutputPath,
+        },
+        TransformResources: {
+          InstanceCount: numInstances,
+          InstanceType: awsInstanceType,
+        },
+      }
+      logJson("transformParams", transformParams)
+
+      const transformResponse = await client.createTransformJob(transformParams).promise()
+      logJson("transformResponse", transformResponse)
+
+      // return success response
+      return new Hub.ActionResponse({ success: true })
+
+    } catch (err) {
+      winston.error(err)
+      return new Hub.ActionResponse({ success: false, message: err.message })
+    }
+  }
+
+  async form(request: Hub.ActionRequest) {
+    winston.debug("form")
+    logJson("request.params", request.params)
+
+    const buckets = await this.listBuckets(request)
+    if (! Array.isArray(buckets)) {
+      throw "Unable to retrieve buckets"
+    }
+    logJson("buckets", buckets)
+
+    const models = await this.listModels(request)
+    if (! Array.isArray(models)) {
+      throw "Unable to retrieve models"
+    }
+    logJson("models", models)
+
+    const form = new Hub.ActionForm()
+    form.fields = [
+      {
+        label: "Model",
+        name: "modelName",
+        required: true,
+        options: models.map((model: any) => {
+          return {
+            name: model.ModelName,
+            label: model.ModelName,
+          }
+        }),
+        type: "select",
+        description: "The S3 bucket where SageMaker input training data should be stored",
+      },
+      {
+        label: "Strip Columns",
+        name: "numStripColumns",
+        required: true,
+        options: [
+          { name: "0", label: "None" },
+          { name: "1", label: "First Column" },
+          { name: "2", label: "First & Second Column" },
+        ],
+        type: "select",
+        default: "0",
+        description: "Columns to remove before running inference task. Columns must be first or second column in the data provided. Use this to remove key, target variable, or both.",
+      },
+      {
+        label: "Output Bucket",
+        name: "bucket",
+        required: true,
+        options: buckets.map((bucket) => {
+          return {
+            name: bucket.Name!,
+            label: bucket.Name!,
+          }
+        }),
+        type: "select",
+        // default: buckets[0].Name, // DNR
+        default: "looker-marketing-analysis", // DNR
```

Comment:
Any particular reason to hardcode this one? Forgive me for not being familiar with this

Reply:
my mistake. these are marked DNR = do not release. removing.